pkg/apis/networking/types.go

```diff
@@ -699,3 +700,58 @@ type ClusterCIDRList struct {
 	// items is the list of ClusterCIDRs.
 	Items []ClusterCIDR
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// IPAddress represents an IP used in a Kubernetes cluster, commonly associated to a Service ClusterIP.
+// An IP address can be represented in different formats, to guarantee the uniqueness of the IP,
+// the name of the object is the IP address in canonical format, four decimal digits separated
+// by dots suppressing leading zeros for IPv4 and the representation defined by RFC 5952 for IPv6.
+// Valid: 192.168.1.5 or 2001:db8::1
+// Invalid: 10.01.2.3 or 2001:db8:0:0:0::1
+type IPAddress struct {
+	metav1.TypeMeta
+	// +optional
+	metav1.ObjectMeta
+	// +optional
+	Spec IPAddressSpec
+}
+
+// IPAddressSpec describe the attributes in an IP Address,
+type IPAddressSpec struct {
+	// ParentRef references the resources that an IPAddress wants to be attached to.
+	// An IPAddress must reference a parent object.
+	// +required
+	ParentRef *ParentReference
+}
+type ParentReference struct {
```

Comment:
at some point of time we will have an idiomatic way across the board to do that. here is hoping..

Reply:
https://github.com/kubernetes/kubernetes/pull/115075#discussion_r1135721159